mobile/lib/routing/auth_guard.dart

```diff
@@ -4,29 +4,40 @@ import 'package:auto_route/auto_route.dart';
 import 'package:flutter/foundation.dart';
 import 'package:immich_mobile/routing/router.dart';
 import 'package:immich_mobile/shared/services/api.service.dart';
+import 'package:logging/logging.dart';
 import 'package:openapi/api.dart';
 
 class AuthGuard extends AutoRouteGuard {
   final ApiService _apiService;
   AuthGuard(this._apiService);
   @override
   void onNavigation(NavigationResolver resolver, StackRouter router) async {
+    var log = Logger("AuthGuard");
+
+    resolver.next(true);
+
     try {
       var res = await _apiService.authenticationApi.validateAccessToken();
-      if (res != null && res.authStatus) {
-        resolver.next(true);
-      } else {
+      if (res == null || res.authStatus != true) {
+        // If the access token is invalid, take user back to login
+        log.info("User token is invalid. Redirecting to login");
         router.replaceAll([const LoginRoute()]);
       }
     } on ApiException catch (e) {
       if (e.code == HttpStatus.badRequest &&
           e.innerException is SocketException) {
         // offline?
-        resolver.next(true);
+        log.info(
```

Comment:
With level info this will be logged very frequently. I'd use fine log level or similar